Add initial version of kanel-enum-tables extension

zimme opened the pull request:
I rebased #350 on main as it was auto closed with the switch from master to main.

I haven't done any code changes and I will try and go over it. Thought I'd open it again so I could get feedback.

zimme commented:
Should be brought up to date with the current code base. I'm a bit unsure about the dependencies of the package. Feedback much appreciated.

kristiandupont commented:
This is nice! I don't have a setup for testing this locally, so I am going to assume that you will :-)

zimme commented:
Just tried with kanel-kysely as well. It doesn't play very well as it does quite a lot of rewriting of the generated types.

kristiandupont commented:
Yeah, see also [this comment](https://github.com/kristiandupont/kanel/issues/571#issuecomment-2642598884) -- the right thing to do at this point is to rewrite things. But I won't have the time to do that any time soon. 

Also, I need to fix the linter issue which is marking every PR as red. Just ignore it.

zimme commented:
Yeah, we're using kysely-codegen anyway for those types. So it's a non issue for me. Just wanted to test it.

Will add a note in the readme that this extension isn't tested with anything other than plain kanel.

But a quick code review for anything obviously bad or general improvements is welcome.

kristiandupont commented on packages/kanel-enum-tables/src/enumTablesPreRenderHook.ts:
```diff
@@ -0,0 +1,191 @@
+    );
+
+    // Extract @enumName from the table comment
+    const enumNameMatch = table.comment.match(/@enumName\s+(?<enumName>\S+)/);
```
You could use [tagged-comment-parser](https://www.npmjs.com/package/tagged-comment-parser) like I do for the type tags. Not a big deal, but it would give you a bit more flexibility if you want to expand the syntax in the future.

zimme replied:
Makes sense, will switch.

kristiandupont reviewed (approved):
Everything looks good to me, it seems that you managed to mirror the quirky style I have ended up adopting in this library! Nicely done! :-)

kristiandupont commented:
I am sorry that this didn't make it all the way through, I feel that I caused that. If you want to complete it, let me know!

zimme commented:
It's on me as well, I said I'd add some extra documentation which I haven't done yet, etc.

zimme commented:
Finally got around to updating this! Here's what changed:

- Switched to `tagged-comment-parser` as suggested. One caveat: the parser doesn't support newline-separated tags (like Postgraphile's `E'@enum\n@enumName Foo'` format), so I parse each line individually and merge the tags. Both Postgraphile's space format and the colon format work.
- Added `kanel-kysely` compatibility — the hook now detects if kysely has already processed the file and updates `ColumnType<>` wrappers accordingly. Works in either hook ordering.
- Fixed several bugs: broken export path, wrong `resolveType` call signature, missing DB connection cleanup, no guard for optional `generateIdentifierType`.
- Expanded the README with smart comment documentation and kysely usage examples.
- Rebased on latest `main`.

Ready for another look when you have time!

zimme commented:
I intentionally put it in this repo hoping it would be more of a community project moving forward.

So I'm ok with you releasing it.
